compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -144,7 +144,10 @@ private void performCodeGen() {
 
         List<Diagnostic> diagnostics = new ArrayList<>();
         for (ModuleContext moduleContext : pkgResolution.topologicallySortedModuleList()) {
-            moduleContext.generatePlatformSpecificCode(compilerContext, this);
+            // We can't generate backend code when one of its dependencies have errors.
+            if (diagnostics.isEmpty()) {
+                moduleContext.generatePlatformSpecificCode(compilerContext, this);
```

Comment:
Shouldn't we check specifically, for errors here?. Having warnings shouldn't stop this.

Reply:
Fixed